test/integration2/migrate/migration-lifecycle-integration.spec.js

```diff
@@ -0,0 +1,303 @@
+'use strict';
+const sinon = require('sinon');
+const chai = require('chai');
+chai.use(require('chai-as-promised'));
+const { expect } = chai;
+
+const path = require('path');
+const rimraf = require('rimraf');
+const logger = require('../../integration/logger');
+const { getAllDbs, getKnexForDb } = require('../util/knex-instance-provider');
+
+describe('Migrations Lifecycle Hooks', function () {
+  getAllDbs().forEach((db) => {
+    describe(db, () => {
+      let knex;
+
+      // Force clean slate before each test
+      beforeEach(async () => {
+        rimraf.sync(path.join(__dirname, './migration'));
```

Comment:
Why sync?

Reply:
Not 100% sure - this setup block was copied from `migration-integration.spec.js` and I chose not to change anything.